component/discovery/kubernetes/k8s.go

```diff
@@ -0,0 +1,136 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/component/metrics/scrape"
+	promk8s "github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+)
+
+// TODO: cpeterson: use defaults from here for hcl default
+var _ = promk8s.DefaultSDConfig
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.k8s",
+		Args:    SDConfig{},
+		Exports: Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(SDConfig))
+		},
+	})
+}
+
+type SDConfig struct {
```

Comment:
Do we want to apply some defaults and validation checks like [Prometheus does](https://github.com/prometheus/prometheus/blob/d9c650b92e0a52a39151792ef3697825e20589c9/discovery/kubernetes/kubernetes.go#L173-L174) in a DecodeHCL method? Or will the errors gracefully float up to our layer?

Reply:
We definitely do.